adapta/utils/_common.py

```diff
@@ -182,3 +182,13 @@ def map_column_names(
     default_values = {k: v for (k, v) in default_values.items() if k not in dataframe.columns}
     dataframe[list(default_values.keys())] = list(default_values.values())
     return dataframe
+
+
+def is_optional(type_: Type) -> bool:
```

Comment:
i'd suggest moving this to a subpackage like `adapta.utils.python_typing._functions`